lobby_server.yaml

```diff
@@ -14,7 +14,7 @@
 - version: "1.10"
   host: lobby.triplea-game.org
   port: 3304
-  http_server_uri: lobby.triplea-game.org:4567
+  http_server_uri: "http://lobby.triplea-game.org:4567"
```

Comment:
We really should configue nginx on this server, so we can use https right from the start.
We should be able to use the config from nodebb, I can look that one up

Reply:
Interesting, I was not considering nginx to be front app servers necessarily, the plan was to get the app servers to be https. I wonder if we still should do that. I was suspecting adding in Https would be the time when Spark showed strains and DropWizard would likely be introduced to replace. 

This also looks interesting: https://github.com/zivver/dropwizard-ratelimit

I think we have a few tasks ahead of us:
- get the app server to be HTTPS
- keep considering nginx
  - get nginx to be https

Reply:
Just looked up the docs, and it seems that both are fairly easy to setup for HTTPS:
For Dropwizard only the config file needs to be updated, which also allows more complex settings to be configured:
https://www.dropwizard.io/1.3.5/docs/manual/configuration.html#https
For spark, setting up HTTPS is super straightforwards as well, but there's a catch:
As explained in http://sparkjava.com/documentation#examples-and-faq it's a one-liner to setup HTTPS, but there's no simple way to configure cipher suites, we'd need to setup a custom jetty connector.
https://github.com/perwendel/spark/issues/925#issuecomment-334898269

In any case we'd have to generate a java keystore file (`.jks`) though, which isn't that straightforward to do using let's encrypt. We'd have to use the "standalone certificate generation variant" and setup a post trigger for that, which converts the (`.pem`) private key and the certificate chain into a valid java keystore file automatically.

In a way I also trust nginx more than jetty (which both frameworks use), probably because I'm a little biased towards the things I'm familiar with. (Using the reverse proxy with NodeBB has worked very well so far, HTTPS configuration could be done independently of NodeBB which is important in a world where more and more applications run on the web.)